lib/puppet/util/windows/api_types.rb

```diff
@@ -165,6 +165,9 @@ def read_com_memory_pointer(&block)
   # https://blogs.msdn.com/b/oldnewthing/archive/2011/03/28/10146459.aspx
   FFI.typedef :int32, :win32_bool
 
+  # The BOOLEAN data type (different to BOOL) is only 1 byte.
+  FFI.typedef :int8, :boolean
```

Comment:
This is not quite right as the Win32 `BYTE` is unsigned and the FFI `:int8` is signed per https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751%28v=vs.85%29.aspx.

A little further in the file we define `:byte` (and map it to the FFI `:uchar`) - so I'd suggest we use that same mapping... https://github.com/ffi/ffi/wiki/Types

```ruby
FFI.typedef :uchar, :boolean
```